src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlTransformRoutedEvent.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using XamlX.Ast;
+using XamlX.Emit;
+using XamlX.IL;
+using XamlX.Transform;
+using XamlX.TypeSystem;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers;
+
+internal class AvaloniaXamlIlTransformRoutedEvent : IXamlAstTransformer
+{
+    public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode node)
+    {
+        if (node is XamlAstNamePropertyReference prop
+            && prop.TargetType is XamlAstClrTypeReference targetRef
+            && prop.DeclaringType is XamlAstClrTypeReference declaringRef)
+        {
+            var xkt = context.GetAvaloniaTypes();
+            var interactiveType = xkt.Interactivity.Interactive;
+            var routedEventType = xkt.Interactivity.RoutedEvent;
+            var AddHandlerT = xkt.Interactivity.AddHandlerT;
+
+            if (interactiveType.IsAssignableFrom(targetRef.Type))
+            {
+                var eventName = $"{prop.Name}Event";
+                if (declaringRef.Type.GetAllFields().FirstOrDefault(f => f.IsStatic && f.Name == eventName) is { } eventField)
```

Comment:
It has the same problem, as in my EventSetter PR, since event name might not necessary match field name+"event".

Reply:
I think we should give ourselves a metric. In the transaction period, an analyzer could be created that signals the lack of convection as happens for avalonia properties.